### DISCUSSION TITLE
[WASM + .NET 8] Status of FullAOT?

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#14500] [Q&A] [answered]
With .NET 8 I gave `FullAOT `another try but ran into the errors below.

I liked to understand where is the limitation (still) is coming from:
- is this something Microsoft still needed to work on?
- is this something where Microsoft already provided the groundworks with .NET8 but UNO still has some work to do?
- else?

Thanks in advance for shedding some light.

<img width="867" alt="image" src="https://github.com/unoplatform/uno/assets/26179242/01e750fd-e68a-4543-ab55-5befe1303588">

**Comment by jeromelaban** [comment, marked as answer]
Microsoft is not working on FullAOT, and since it is limited by WebAssembly's featureset, it is not possible at this time.

**Reply by DierkDroth** [reply]
Thanks for you feedback @jeromelaban 

That's odd. There still are scenarios in my app where UNO WASM is significantly slower than WindowsAppSDK (native). I was hoping to find an 'easy solution' by switching to `FullAOT`.

**Reply by jeromelaban** [reply]
It's not odd. If Wasm does not provide a feature, the runtime team cannot invent it. 

The only piece that brings closer to FullAOT is the Jiterpreter, but even that does not compensate for the missing underlying features.

If you have scenarios that are slower that they should be, it could be that the AOT compiler is not generating proper code. It's best to create minimal repros (with profiler traces) and post those on the dotnet/runtime repo for them to look at.

**Reply by DierkDroth** [reply]
Thanks @jeromelaban. I'll keep that mind. At this point there just is not the bandwidth to chase performance issues, but I might revisit that later.

**Reply by limeniye** [reply]
Additional: https://www.youtube.com/watch?v=TnrDbkN28yg&t=1020s&ab_channel=UnoPlatform
